app/app.go

```diff
@@ -100,6 +100,9 @@ func (a *App) InitServer() {
 			})
 		}
 		a.srv.RunJobs()
+		if a.IsLeader() {
```

Comment:
The problem of this is, on leader change, it doesn't create the new task. For example, if the leader not gets restarted, and other node takes the leadership, is not going to unset the DND statuses.

Reply:
should we start task on all servers or is there any way to know when leader change event happens?

Reply:
There is a function in `app/cluster.go` called `AddClusterLeaderChangedListener` that allows you to register a cluster leader change, that should do the trick. You can see an example of the usage of this in the line 420 of the `app/server.go` file.

Reply:
buy the way, this logic fits better in the `app/server.go:NewServer` function, than in the `app/app.go:InitServer` function.

Reply:
but in new server we don't have App available so we can't access UpdateDNDStatusOfUsers method